### ISSUE TITLE
How to speed up training

### ISSUE THREAD

**Issue by idesign2018** [bmaltais/kohya_ss#3467]
My computer has an i9-14900KF CPU, an RTX 5090 GPU with 32GB of VRAM, 128GB of RAM, and a 2TB SSD. I'm using the kohya_ss v25.2.1 trainer and want to train a Flux.1 model. How should I configure it to utilize my computer's full performance for training? The problem I'm encountering today is that the VRAM usage is around 31.3 MHz, but the GPU utilization is only around 30%, and it's very slow. Does kohya_ss not support the RTX 5090 graphics card?

<img width="1886" height="795" alt="Image" src="https://github.com/user-attachments/assets/d212c0bb-ce0a-4581-8b6d-1d13f27d32ff" />

**Comment by AbstractEyes** [comment]
Most often the problems revolve around the speed of data being fed in, in this case it is likely related to the T5 being on the cpu. Freeze the t5 and cache the clip latents. Freeze the AE and cache the image latents. Train the CLIP_L gently, it's small and you don't need that cached so it can train a little - but not too much.

I suggest increasing the batch size and reducing or removing the block swap entirely with the T5 latents cached. Block swap is essentially the model moving pieces to and from the gpu.

If your cuda isn't the correct version, you can potentially have bf16 conversion lag. fp8 e4m3fn is considered faster than bf16, but the dtype differences can cause cache misses which compound lag.

T5xxl is the primary lag inducer with flux, so if the t5 is on cpu it's going to run slowly, especially if you have the clip learn rate above 0.

An A100 40 gig can barely house the entirety of the T5xxl on the gpu with the correct dtype, so a pc with the flux unet loaded isn't going to be able to handle it.

**Comment by idesign2018** [comment]
> Most often the problems revolve around the speed of data being fed in, in this case it is likely related to the T5 being on the cpu. Freeze the t5 and cache the clip latents. Freeze the AE and cache the image latents. Train the CLIP_L gently, it's small and you don't need that cached so it can train a little - but not too much.
> 
> I suggest increasing the batch size and reducing or removing the block swap entirely with the T5 latents cached. Block swap is essentially the model moving pieces to and from the gpu.
> 
> If your cuda isn't the correct version, you can potentially have bf16 conversion lag. fp8 e4m3fn is considered faster than bf16, but the dtype differences can cause cache misses which compound lag.
> 
> T5xxl is the primary lag inducer with flux, so if the t5 is on cpu it's going to run slowly, especially if you have the clip learn rate above 0.
> 
> An A100 40 gig can barely house the entirety of the T5xxl on the gpu with the correct dtype, so a pc with the flux unet loaded isn't going to be able to handle it.

Do I need to change the following two parameters to "0"?
Text Encoder learning rate: 0
T5XXL learning rate  T5XXL: 0
---
I couldn't find the following settings options in the Kohya SS GUI: How do I set these parameters?
Freeze T5，Freeze AE，cache CLIP latents，cache image latents，block swap
---
Which version of CUDA should I install for my 5090 graphics card? I currently have the latest CUDA version.
